php-http/httplug-bundle/1.6/config/packages/httplug.yaml

```diff
@@ -12,3 +12,7 @@ httplug:
             plugins:
                 - 'httplug.plugin.content_length'
                 - 'httplug.plugin.redirect'
+                - header_defaults:
+                    headers:
```

Comment:
i don't see what would be wrong with this line, its allowed by the configuration: https://github.com/php-http/HttplugBundle/blob/v1.6.0/DependencyInjection/Configuration.php#L319

Reply:
What happens with the following?

```yaml
httplug:
    plugins:
        redirect:
            preserve_header: true

    discovery:
        client: 'auto'

    clients:
        app:
            http_methods_client: true
            plugins:
                - 'httplug.plugin.content_length'
                - 'httplug.plugin.redirect'
                - header_defaults:
                      headers:
                          "X-Httplug": "This is just an example."
                - 'httplug.plugin.retry'
```

Reply:
That’s is what we started with. Right? @dbu

Reply:
I moved the dot of the header value into the quotes.

Reply:
ah damn, i did not see that dot :man_facepalming: 
yep, that was the problem

Reply:
The error message was indeed a bit misleading. http://yaml-online-parser.appspot.com/ is a good place to quickly verify some YAML string.